g711.py

```diff
@@ -0,0 +1,107 @@
+__all__ = ['alaw_encode', 'alaw_decode', 'ulaw_encode', 'ulaw_decode', 'g711_read', 'g711_write', 'open']
+
+from _g711.lib import alaw_encode as _alaw_encode, alaw_decode as _alaw_decode, ulaw_encode as _ulaw_encode, ulaw_decode as _ulaw_decode
+
+import builtins
+
+from cffi import FFI
+ffi = FFI()
+
+
+class Error(Exception):
+    pass
+
+
+def _apply(bytes_in, func, encode):
+    if encode and len(bytes_in) % 2 != 0:
+        raise Error("Number of bytes to be encoded must be even as each sample requires two bytes")
+    div = 2 if encode else 1
+    in_buf = ffi.new("int16_t[]" if encode else "int8_t[]", len(bytes_in) // div)
+    ot_buf = ffi.new("int8_t[]" if encode else "int16_t[]", len(bytes_in) // div)
+
+    ffi.buffer(in_buf)[:] = bytes_in
```

Comment:
I think there's a way to do this without a copy but we might revisit this later if we need, probably fine for now as-is.